insights/client/__init__.py

```diff
@@ -68,94 +68,21 @@ def test_connection(self):
-        if new_egg and not skip_verify:
-            verification = self.verify(new_egg, gpg_key)
-            logger.debug('Core was verified: %s', verification)
-
-        # Need to install the new Core here
-        if new_egg and verification['gpg']:
-            installation = self.install(new_egg)
-            logger.debug('Core installation: %s', installation)
-            # Return 42 to the wrapper
-            # Indicates we want to stop execution and start the new Client
-            if installation['success']:
-                return 42
-            else:
-                logger.debug('There was an error installing the new core.')
-        else:
-            logger.debug('New egg was not retrieved or was retrieved but failed verification.')
-            logger.debug('Egg retrieval: %s', new_egg)
-            logger.debug('Verification: %s', verification)
-
-        # Register
-        is_registered = self.get_registration_information()['is_registered']
-        logger.debug('System is registered: %s', is_registered)
-        if not config['offline'] and not is_registered:
-            registration = self.register(force_register)
-            is_registered = registration['registration']['status']
-            logger.debug('Registration response: %s', registration)
-            logger.debug('System is now registered: %s', is_registered)
-
-        # Collect things
-        if verification:
-            logger.debug('New Core was verified. Collecting information.')
-
-            if update_rules:
-                logger.debug("Updating rules.")
-                self.update_rules()  # won't be needed after we move to new egg format
-
-            results = self.collect()
-            logger.debug('Results: %s', results)
-        else:
-            logger.debug('New Core was not verified, not collecting information.')
-            results = False
-
-        # Upload things
-        if not skip_upload and not config['no_upload'] and not config['offline'] and is_registered:
-            logger.debug('Not skipping upload, or running offline.')
-            logger.debug('System is registered, proceeding with upload.')
-            upload_results = self.upload(results)
-            logger.debug('Upload results: %s', upload_results)
-        else:
-            logger.debug('Skipping upload, running offline, or system is not properly registered.')
-            logger.debug('Skipping upload.')
-            logger.debug('Insights results: %s', results)
-            return results
-
-    def fetch(self,
-              egg_url=constants.egg_path,
-              force=False,
-              verbose=False):
+    def fetch(self, **kwargs):
         """
+            parms:
+                core_url (str): URL to retrieve egg from
+                force (bool): force fetch egg (dont check etags)
             returns (str): path to new egg.  None if no update.
         """
         # was a custom egg url passed in?
         if config['core_url']:
```

Comment:
Why not just combine this into one line:

```
egg_url = kwargs.get('core_url', config['core_url'])
```

And just make sure the default in the config for `core_url` is `constants.egg_path`